File: src/test/java/com/melonbase/microquark/repo/KantonTest.java
package com.melonbase.microquark.repo;

import com.melonbase.microquark.repo.data.Kanton;
import org.junit.jupiter.api.Test;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class KantonTest {

  @Test
  void correctNumberOfCantons() {
    assertEquals(26, Kanton.values().length);
  }

  @Test
  void correctTotalInhabitants() {
    assertEquals(8_551_744, Stream.of(Kanton.values())
        .mapToLong(Kanton::getEinwohner)
        .sum());
  }
}

File: src/test/kotlin/com/melonbase/microquark/repo/KantonTest.kt
package com.melonbase.microquark.repo

import com.melonbase.microquark.repo.data.Kanton
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class KantonTest {

  @Test
  fun correctNumberOfCantons() {
    assertEquals(26, Kanton.values().size)
  }

  @Test
  fun correctTotalInhabitants() {
    assertEquals(8_551_744, Kanton.values().sumOf { it.einwohner })
  }
}